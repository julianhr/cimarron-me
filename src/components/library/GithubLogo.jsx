import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'

import githubLogoPng from '~/assets/img/github_logo.png'


const Img = styled.img`
  width: 60px;
  padding-left: 5px;
`

function GithubLogo({ url, styles }) {
  return (
    <a
      href={url} 
      target='_blank'
      rel='noopener noreferrer'
      css={styles.a}
    >
      <Img
        src={githubLogoPng}
        alt='GitHub'
        css={styles.img}
      />
    </a>
  )
}

GithubLogo.propTypes = {
  url: PropTypes.string,
  styles: PropTypes.object,
}

GithubLogo.defaultProps = {
  styles: {}
}

export default GithubLogo
