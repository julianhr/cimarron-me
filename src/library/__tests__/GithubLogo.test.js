import React from 'react'
import { mount } from 'enzyme'
import emotionSerializer from 'jest-emotion'

import GithubLogo from '../GithubLogo'


const testProps = {
  url: 'http://test.com',
  styles: { 
    span: {
      color: 'orange'
    },
    a: {
      color: 'red'
    },
    img: {
      color: 'green'
    },
  },
}

describe('GithubLogo', () => {
  describe('required props', () => {
    it('url', () => {
      const props = {...testProps}
      delete props['url']
      expect(() => <GithubLogo {...props} />).toThrow()
    })
  })

  describe('snapshots', () => {
    beforeAll(() => {
      expect.addSnapshotSerializer(emotionSerializer)
    })

    it('with all props', () => {
      const wrapper = mount(<GithubLogo {...testProps} />)
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })
    
    it('without styles prop', () => {
      const wrapper = mount(<GithubLogo url={testProps.url} />)
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })
  })
})
