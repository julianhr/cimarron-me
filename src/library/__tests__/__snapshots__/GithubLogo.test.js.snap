// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubLogo snapshots with all props 1`] = `
.emotion-4 {
  color: orange;
}

.emotion-3 {
  color: red;
}

.emotion-2 {
  color: green;
}

.emotion-0 {
  width: 60px;
  color: green;
}

<span
  className="emotion-4"
>
  <a
    className="emotion-3"
    href="http://test.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <Img
      alt="GitHub"
      className="emotion-2"
      src="test-file-stub"
    >
      <img
        alt="GitHub"
        className="emotion-0 emotion-1"
        src="test-file-stub"
      />
    </Img>
  </a>
</span>
`;

exports[`GithubLogo snapshots without styles prop 1`] = `
.emotion-0 {
  width: 60px;
}

<span>
  <a
    href="http://test.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <Img
      alt="GitHub"
      src="test-file-stub"
    >
      <img
        alt="GitHub"
        className="emotion-0 emotion-1"
        src="test-file-stub"
      />
    </Img>
  </a>
</span>
`;
